I keep getting the following error running the **test-acceptance-api** tests locally, any ideas?

```
Nothing to install or update
Generating autoload files
composer/package-versions-deprecated: Generating version class...
composer/package-versions-deprecated: ...done generating version class
61 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
./tests/acceptance/run.sh --type api
Script path: /home/robertp/p/core/tests/acceptance
Using php inbuilt server for running scenario ...
8080
6151
8180
6152
[Tue Feb 23 17:12:15 2021] PHP 7.4.3 Development Server (http://localhost:8180) started
[Tue Feb 23 17:12:15 2021] PHP 7.4.3 Development Server (http://localhost:8080) started
[Tue Feb 23 17:12:20 2021] 127.0.0.1:38312 Accepted
[Tue Feb 23 17:12:20 2021] 127.0.0.1:38312 [302]: POST /ocs/v2.php/apps/testing/api/v1/occ
[Tue Feb 23 17:12:20 2021] 127.0.0.1:38312 Closing
-:2: parser error : Start tag expected, '<' not found

^
[Tue Feb 23 17:12:20 2021] 127.0.0.1:38314 Accepted
[Tue Feb 23 17:12:20 2021] 127.0.0.1:38314 [302]: POST /ocs/v2.php/apps/testing/api/v1/occ
[Tue Feb 23 17:12:20 2021] 127.0.0.1:38314 Closing
-:2: parser error : Start tag expected, '<' not found

^
Could not set mail_domain on http://localhost:8080/ocs/v2.php/apps/testing/api/v1/occ. Result:
''
[Tue Feb 23 17:12:20 2021] 127.0.0.1:38316 Accepted
[Tue Feb 23 17:12:20 2021] 127.0.0.1:38316 [302]: POST /ocs/v2.php/apps/testing/api/v1/occ
[Tue Feb 23 17:12:20 2021] 127.0.0.1:38316 Closing
-:2: parser error : Start tag expected, '<' not found

^
runsh: Exit code of main run:
make: *** [Makefile:207: test-acceptance-api] Error 1```

--- a/apps/provisioning_api/appinfo/routes.php
+++ b/apps/provisioning_api/appinfo/routes.php
@@ -53,7 +53,7 @@ API::register('post', '/cloud/users/{userid}/groups', [$users, 'addToGroup'], 'p
 API::register('delete', '/cloud/users/{userid}/groups', [$users, 'removeFromGroup'], 'provisioning_api', API::SUBADMIN_AUTH);
 API::register('post', '/cloud/users/{userid}/subadmins', [$users, 'addSubAdmin'], 'provisioning_api', API::ADMIN_AUTH);
 API::register('delete', '/cloud/users/{userid}/subadmins', [$users, 'removeSubAdmin'], 'provisioning_api', API::ADMIN_AUTH);
-API::register('get', '/cloud/users/{userid}/subadmins', [$users, 'getUserSubAdminGroups'], 'provisioning_api', API::ADMIN_AUTH);
+API::register('get', '/cloud/users/{userid}/subadmins', [$users, 'getUserSubAdminGroups'], 'provisioning_api', API::USER_AUTH);
 
 // Groups
 $groups = new Groups(

--- a/apps/provisioning_api/lib/Users.php
+++ b/apps/provisioning_api/lib/Users.php
@@ -628,22 +628,40 @@ class Users {
 	 * @return Result
 	 */
 	public function getUserSubAdminGroups($parameters) {
-		$user = $this->userManager->get($parameters['userid']);
-		// Check if the user exists
-		if ($user === null) {
-			return new Result(null, 101, 'User does not exist');
+		$targetUserId = $parameters['userid'];
+		$targetUser = $this->userManager->get($targetUserId);
+
+		// Check if user is logged in
+		$currentLoggedInUser = $this->userSession->getUser();
+		if ($currentLoggedInUser === null) {
+			return new Result(null, API::RESPOND_UNAUTHORISED);
 		}
 
-		// Get the subadmin groups
-		$groups = $this->groupManager->getSubAdmin()->getSubAdminsGroups($user);
-		foreach ($groups as $key => $group) {
-			$groups[$key] = $group->getGID();
-		}
+		// Check if same user or subadmin or admin
+		$subAdminManager = $this->groupManager->getSubAdmin();
+		if ($targetUserId === $currentLoggedInUser->getUID() ||
+			$subAdminManager->isUserAccessible($currentLoggedInUser, $targetUser) ||
+			$this->groupManager->isAdmin($currentLoggedInUser->getUID())) {
 
-		if (!$groups) {
-			return new Result(null, 102, 'Unknown error occurred');
+			// Check if the user exists
+			if ($targetUser === null) {
+				return new Result(null, 101, 'User does not exist');
+			}
+
+			// Get the subadmin groups
+			$groups = $this->groupManager->getSubAdmin()->getSubAdminsGroups($targetUser);
+			foreach ($groups as $key => $group) {
+				$groups[$key] = $group->getGID();
+			}
+
+			if (!$groups) {
+				return new Result(null, 102, 'Unknown error occurred');
+			} else {
+				return new Result($groups);
+			}
 		} else {
-			return new Result($groups);
+			// No rights
+			return new Result(null, 997);
 		}
 	}
 

--- a/apps/provisioning_api/lib/Users.php
+++ b/apps/provisioning_api/lib/Users.php
@@ -628,8 +628,6 @@ class Users {
 	 * @return Result
 	 */
 	public function getUserSubAdminGroups($parameters) {
-		$targetUserId = $parameters['userid'];
-		$targetUser = $this->userManager->get($targetUserId);
 
 		// Check if user is logged in
 		$currentLoggedInUser = $this->userSession->getUser();
@@ -637,31 +635,32 @@ class Users {
 			return new Result(null, API::RESPOND_UNAUTHORISED);
 		}
 
+		$currentLoggedInUserId = $currentLoggedInUser->getUID();
+
+		$targetUserId = $parameters['userid'];
+		$targetUser = $this->userManager->get($targetUserId);
+
+		// Check if the user exists
+		if ($targetUser === null) {
+			return new Result(null, API::RESPOND_NOT_FOUND, 'The requested user could not be found');
+		}
+
 		// Check if same user or subadmin or admin
 		$subAdminManager = $this->groupManager->getSubAdmin();
-		if ($targetUserId === $currentLoggedInUser->getUID() ||
-			$subAdminManager->isUserAccessible($currentLoggedInUser, $targetUser) ||
-			$this->groupManager->isAdmin($currentLoggedInUser->getUID())) {
-
-			// Check if the user exists
-			if ($targetUser === null) {
-				return new Result(null, 101, 'User does not exist');
-			}
+		if ($targetUserId === $currentLoggedInUserId
+			|| $subAdminManager->isUserAccessible($currentLoggedInUser, $targetUser)
+			|| $this->groupManager->isAdmin($currentLoggedInUserId)) {
 
 			// Get the subadmin groups
-			$groups = $this->groupManager->getSubAdmin()->getSubAdminsGroups($targetUser);
+			$groups = $subAdminManager->getSubAdminsGroups($targetUser);
 			foreach ($groups as $key => $group) {
 				$groups[$key] = $group->getGID();
 			}
 
-			if (!$groups) {
-				return new Result(null, 102, 'Unknown error occurred');
-			} else {
-				return new Result($groups);
-			}
+			return new Result($groups ? $groups : []);
 		} else {
-			// No rights
-			return new Result(null, 997);
+			// No permission to access users groups
+			return new Result(null, API::RESPOND_UNAUTHORISED);
 		}
 	}
 

--- a/apps/provisioning_api/tests/UsersTest.php
+++ b/apps/provisioning_api/tests/UsersTest.php
@@ -2506,20 +2506,42 @@ class UsersTest extends OriginalTest {
 		$this->assertEquals($expected, $this->api->removeSubAdmin(['userid' => 'ExistingUser', '_delete' => ['groupid' => 'GroupToDeleteFrom']]));
 	}
 
+	public function testGetUserSubAdminGroupsNoCurrentUserInSession() {
+		$expected = new Result(null, 997);
+		$this->assertEquals($expected, $this->api->getUserSubAdminGroups(['userid' => 'RequestedUser']));
+	}
+
 	public function testGetUserSubAdminGroupsNotExistingTargetUser() {
+		$currentUser = $this->createMock(IUser::class);
+		$this->userSession
+			->expects($this->once())
+			->method('getUser')
+			->will($this->returnValue($currentUser));
+		$currentUser
+			->expects($this->once())
+			->method('getUID')
+			->will($this->returnValue('RequestedUser'));
 		$this->userManager
 			->expects($this->once())
 			->method('get')
 			->with('RequestedUser')
 			->will($this->returnValue(null));
 
-		$expected = new Result(null, 101, 'User does not exist');
+		$expected = new Result(null, 998, 'The requested user could not be found');
 		$this->assertEquals($expected, $this->api->getUserSubAdminGroups(['userid' => 'RequestedUser']));
 	}
 
-	public function testGetUserSubAdminGroupsWithGroups() {
+	public function testGetUserSubAdminGroupsWithGroupsAsSameUser() {
 		$targetUser = $this->createMock(IUser::class);
 		$targetGroup = $this->createMock(IGroup::class);
+		$this->userSession
+			->expects($this->once())
+			->method('getUser')
+			->will($this->returnValue($targetUser));
+		$targetUser
+			->expects($this->once())
+			->method('getUID')
+			->will($this->returnValue('RequestedUser'));
 		$targetGroup
 			->expects($this->once())
 			->method('getGID')
@@ -2545,8 +2567,115 @@ class UsersTest extends OriginalTest {
 		$this->assertEquals($expected, $this->api->getUserSubAdminGroups(['userid' => 'RequestedUser']));
 	}
 
-	public function testGetUserSubAdminGroupsWithoutGroups() {
+	public function testGetUserSubAdminGroupsAsSubadmin() {
+		$currentUser = $this->createMock(IUser::class);
 		$targetUser = $this->createMock(IUser::class);
+		$targetGroup = $this->createMock(IGroup::class);
+		$this->userSession
+			->expects($this->once())
+			->method('getUser')
+			->will($this->returnValue($currentUser));
+		$targetGroup
+			->expects($this->once())
+			->method('getGID')
+			->will($this->returnValue('TargetGroup'));
+		$this->userManager
+			->expects($this->once())
+			->method('get')
+			->with('RequestedUser')
+			->will($this->returnValue($targetUser));
+		$subAdminManager = $this->getMockBuilder(SubAdmin::class)
+			->disableOriginalConstructor()->getMock();
+		$subAdminManager
+			->expects($this->once())
+			->method('getSubAdminsGroups')
+			->with($targetUser)
+			->will($this->returnValue([$targetGroup]));
+		$subAdminManager
+			->expects($this->once())
+			->method('isUserAccessible')
+			->with($currentUser, $targetUser)
+			->will($this->returnValue(true));
+		$this->groupManager
+			->expects($this->once())
+			->method('getSubAdmin')
+			->will($this->returnValue($subAdminManager));
+
+		$expected = new Result(['TargetGroup'], 100);
+		$this->assertEquals($expected, $this->api->getUserSubAdminGroups(['userid' => 'RequestedUser']));
+	}
+
+	public function testGetUserSubAdminGroupsAsSubadminUserNotAccessible() {
+		$currentUser = $this->createMock(IUser::class);
+		$targetUser = $this->createMock(IUser::class);
+		$targetGroup = $this->createMock(IGroup::class);
+		$this->userSession
+			->expects($this->once())
+			->method('getUser')
+			->will($this->returnValue($currentUser));
+		$this->userManager
+			->expects($this->once())
+			->method('get')
+			->with('RequestedUser')
+			->will($this->returnValue($targetUser));
+		$subAdminManager = $this->getMockBuilder(SubAdmin::class)
+			->disableOriginalConstructor()->getMock();
+		$subAdminManager
+			->expects($this->once())
+			->method('isUserAccessible')
+			->with($currentUser, $targetUser)
+			->will($this->returnValue(false));
+		$this->groupManager
+			->expects($this->once())
+			->method('getSubAdmin')
+			->will($this->returnValue($subAdminManager));
+
+		$expected = new Result(null, 997);
+		$this->assertEquals($expected, $this->api->getUserSubAdminGroups(['userid' => 'RequestedUser']));
+	}
+
+	public function testGetUserSubAdminGroupsWithGroupsAsAdmin() {
+		$targetUser = $this->createMock(IUser::class);
+		$targetGroup = $this->createMock(IGroup::class);
+		$this->userSession
+			->expects($this->once())
+			->method('getUser')
+			->will($this->returnValue($targetUser));
+		$targetGroup
+			->expects($this->once())
+			->method('getGID')
+			->will($this->returnValue('TargetGroup'));
+		$this->userManager
+			->expects($this->once())
+			->method('get')
+			->with('RequestedUser')
+			->will($this->returnValue($targetUser));
+		$subAdminManager = $this->getMockBuilder(SubAdmin::class)
+			->disableOriginalConstructor()->getMock();
+		$subAdminManager
+			->expects($this->once())
+			->method('getSubAdminsGroups')
+			->with($targetUser)
+			->will($this->returnValue([$targetGroup]));
+		$this->groupManager
+			->expects($this->once())
+			->method('getSubAdmin')
+			->will($this->returnValue($subAdminManager));
+		$this->groupManager
+			->expects($this->once())
+			->method('isAdmin')
+			->will($this->returnValue(true));
+
+		$expected = new Result(['TargetGroup'], 100);
+		$this->assertEquals($expected, $this->api->getUserSubAdminGroups(['userid' => 'RequestedUser']));
+	}
+
+	public function testGetUserSubAdminGroupsWithoutAnyGroups() {
+		$targetUser = $this->createMock(IUser::class);
+		$this->userSession
+			->expects($this->once())
+			->method('getUser')
+			->will($this->returnValue($targetUser));
 		$this->userManager
 			->expects($this->once())
 			->method('get')
@@ -2563,8 +2692,12 @@ class UsersTest extends OriginalTest {
 			->expects($this->once())
 			->method('getSubAdmin')
 			->will($this->returnValue($subAdminManager));
+		$this->groupManager
+			->expects($this->once())
+			->method('isAdmin')
+			->will($this->returnValue(true));
 
-		$expected = new Result(null, 102, 'Unknown error occurred');
+		$expected = new Result([], 100);
 		$this->assertEquals($expected, $this->api->getUserSubAdminGroups(['userid' => 'RequestedUser']));
 	}
 

--- a/changelog/unreleased/38281
+++ b/changelog/unreleased/38281
@@ -1,0 +1,3 @@
+Bugfix: Allow all users to see which groups they manage
+
+https://github.com/owncloud/core/pull/38281

--- a/tests/acceptance/features/apiProvisioningGroups-v1/getSubAdminGroups.feature
+++ b/tests/acceptance/features/apiProvisioningGroups-v1/getSubAdminGroups.feature
@@ -26,7 +26,7 @@ Feature: get subadmin groups
     Given user "nonexistentuser" has been deleted
     And group "brand-new-group" has been created
     When the administrator gets all the groups where user "nonexistentuser" is subadmin using the provisioning API
-    Then the OCS status code should be "101"
+    Then the OCS status code should be "998"
     And the HTTP status code should be "200"
     And the API should not return any data
 
@@ -38,15 +38,11 @@ Feature: get subadmin groups
     And user "Alice" has been made a subadmin of group "brand-new-group"
     And user "Alice" has been made a subadmin of group "😅 😆"
     When user "Alice" gets all the groups where user "Alice" is subadmin using the provisioning API
-    # Then the OCS status code should be "100"
-    # And the HTTP status code should be "200"
-    # Then the subadmin groups returned by the API should be
-    #   | brand-new-group |
-    #   | 😅 😆           |
-    Then the OCS status code should be "997"
-    And the HTTP status code should be "401"
-    And the API should not return any data
-
+    Then the OCS status code should be "100"
+    And the HTTP status code should be "200"
+    Then the subadmin groups returned by the API should be
+      | brand-new-group |
+      | 😅 😆          |
 
   Scenario: subadmin of a group should not be able to get subadmin groups of another subadmin user
     Given these users have been created with default attributes and small skeleton files:

--- a/tests/acceptance/features/apiProvisioningGroups-v2/getSubAdminGroups.feature
+++ b/tests/acceptance/features/apiProvisioningGroups-v2/getSubAdminGroups.feature
@@ -26,8 +26,8 @@ Feature: get subadmin groups
     Given user "nonexistentuser" has been deleted
     And group "brand-new-group" has been created
     When the administrator gets all the groups where user "nonexistentuser" is subadmin using the provisioning API
-    Then the OCS status code should be "400"
-    And the HTTP status code should be "400"
+    Then the OCS status code should be "404"
+    And the HTTP status code should be "404"
     And the API should not return any data
 
   @issue-owncloud-sdk-658
@@ -38,15 +38,11 @@ Feature: get subadmin groups
     And user "Alice" has been made a subadmin of group "brand-new-group"
     And user "Alice" has been made a subadmin of group "😅 😆"
     When user "Alice" gets all the groups where user "Alice" is subadmin using the provisioning API
-    # Then the OCS status code should be "200"
-    # And the HTTP status code should be "200"
-    # Then the subadmin groups returned by the API should be
-    #   | brand-new-group |
-    #   | 😅 😆           |
-    Then the OCS status code should be "997"
-    And the HTTP status code should be "401"
-    And the API should not return any data
-
+    Then the OCS status code should be "200"
+    And the HTTP status code should be "200"
+    Then the subadmin groups returned by the API should be
+      | brand-new-group |
+      | 😅 😆          |
 
   Scenario: subadmin of a group should not be able to get subadmin groups of another subadmin user
     Given these users have been created with default attributes and small skeleton files:

